// @starton/sdk/dist/sdk/models/shared/abiobjectdto.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbiObjectDto = exports.TypeT = exports.StateMutability = void 0;
var utils_1 = require("../../../internal/utils");
var abiinputdto_1 = require("./abiinputdto");
var abioutputdto_1 = require("./abioutputdto");
var class_transformer_1 = require("class-transformer");
var StateMutability;
(function (StateMutability) {
    StateMutability["Pure"] = "pure";
    StateMutability["View"] = "view";
    StateMutability["Nonpayable"] = "nonpayable";
    StateMutability["Payable"] = "payable";
})(StateMutability = exports.StateMutability || (exports.StateMutability = {}));
var TypeT;
(function (TypeT) {
    TypeT["Constructor"] = "constructor";
    TypeT["Event"] = "event";
    TypeT["Function"] = "function";
    TypeT["Receive"] = "receive";
    TypeT["Fallback"] = "fallback";
    TypeT["Error"] = "error";
})(TypeT = exports.TypeT || (exports.TypeT = {}));
var AbiObjectDto = /** @class */ (function (_super) {
    __extends(AbiObjectDto, _super);
    function AbiObjectDto() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "anonymous" }),
        __metadata("design:type", Boolean)
    ], AbiObjectDto.prototype, "anonymous", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "constant" }),
        __metadata("design:type", Boolean)
    ], AbiObjectDto.prototype, "constant", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "inputs" }),
        (0, class_transformer_1.Type)(function () { return abiinputdto_1.AbiInputDto; }),
        __metadata("design:type", abiinputdto_1.AbiInputDto)
    ], AbiObjectDto.prototype, "inputs", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], AbiObjectDto.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "outputs" }),
        (0, class_transformer_1.Type)(function () { return abioutputdto_1.AbiOutputDto; }),
        __metadata("design:type", abioutputdto_1.AbiOutputDto)
    ], AbiObjectDto.prototype, "outputs", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "payability" }),
        __metadata("design:type", Boolean)
    ], AbiObjectDto.prototype, "payability", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "stateMutability" }),
        __metadata("design:type", String)
    ], AbiObjectDto.prototype, "stateMutability", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "type" }),
        __metadata("design:type", String)
    ], AbiObjectDto.prototype, "type", void 0);
    return AbiObjectDto;
}(utils_1.SpeakeasyBase));
exports.AbiObjectDto = AbiObjectDto;
