// @starton/sdk/dist/sdk/models/shared/calldto.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallDto = exports.Speed = void 0;
var utils_1 = require("../../../internal/utils");
var customgasdto_1 = require("./customgasdto");
var class_transformer_1 = require("class-transformer");
var Speed;
(function (Speed) {
    Speed["Low"] = "low";
    Speed["Average"] = "average";
    Speed["Fast"] = "fast";
    Speed["Fastest"] = "fastest";
    Speed["Custom"] = "custom";
})(Speed = exports.Speed || (exports.Speed = {}));
var CallDto = /** @class */ (function (_super) {
    __extends(CallDto, _super);
    function CallDto() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "customGas" }),
        (0, class_transformer_1.Type)(function () { return customgasdto_1.CustomGasDto; }),
        __metadata("design:type", customgasdto_1.CustomGasDto)
    ], CallDto.prototype, "customGas", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "functionName" }),
        __metadata("design:type", String)
    ], CallDto.prototype, "functionName", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "gasLimit" }),
        __metadata("design:type", String)
    ], CallDto.prototype, "gasLimit", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "nonce" }),
        __metadata("design:type", Number)
    ], CallDto.prototype, "nonce", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "params" }),
        __metadata("design:type", Array)
    ], CallDto.prototype, "params", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "signerWallet" }),
        __metadata("design:type", String)
    ], CallDto.prototype, "signerWallet", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "speed" }),
        __metadata("design:type", String)
    ], CallDto.prototype, "speed", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "value" }),
        __metadata("design:type", String)
    ], CallDto.prototype, "value", void 0);
    return CallDto;
}(utils_1.SpeakeasyBase));
exports.CallDto = CallDto;
