// @starton/sdk/dist/sdk/models/shared/createwatcherdto.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateWatcherDto = exports.CreateWatcherDtoType = exports.CreateWatcherDtoMetadata = exports.CustomFilters = exports.CustomEventAbi = void 0;
var utils_1 = require("../../../internal/utils");
var class_transformer_1 = require("class-transformer");
var CustomEventAbi = /** @class */ (function (_super) {
    __extends(CustomEventAbi, _super);
    function CustomEventAbi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return CustomEventAbi;
}(utils_1.SpeakeasyBase));
exports.CustomEventAbi = CustomEventAbi;
var CustomFilters = /** @class */ (function (_super) {
    __extends(CustomFilters, _super);
    function CustomFilters() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return CustomFilters;
}(utils_1.SpeakeasyBase));
exports.CustomFilters = CustomFilters;
var CreateWatcherDtoMetadata = /** @class */ (function (_super) {
    __extends(CreateWatcherDtoMetadata, _super);
    function CreateWatcherDtoMetadata() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return CreateWatcherDtoMetadata;
}(utils_1.SpeakeasyBase));
exports.CreateWatcherDtoMetadata = CreateWatcherDtoMetadata;
var CreateWatcherDtoType;
(function (CreateWatcherDtoType) {
    CreateWatcherDtoType["MinedTransaction"] = "MINED_TRANSACTION";
    CreateWatcherDtoType["AddressActivity"] = "ADDRESS_ACTIVITY";
    CreateWatcherDtoType["AddressReceivedNativeCurrency"] = "ADDRESS_RECEIVED_NATIVE_CURRENCY";
    CreateWatcherDtoType["AddressSentNativeCurrency"] = "ADDRESS_SENT_NATIVE_CURRENCY";
    CreateWatcherDtoType["EventTransfer"] = "EVENT_TRANSFER";
    CreateWatcherDtoType["Erc721EventTransfer"] = "ERC721_EVENT_TRANSFER";
    CreateWatcherDtoType["Erc1155EventTransferSingle"] = "ERC1155_EVENT_TRANSFER_SINGLE";
    CreateWatcherDtoType["Erc1155EventTransferBatch"] = "ERC1155_EVENT_TRANSFER_BATCH";
    CreateWatcherDtoType["EventMint"] = "EVENT_MINT";
    CreateWatcherDtoType["EventApproval"] = "EVENT_APPROVAL";
    CreateWatcherDtoType["EventCustom"] = "EVENT_CUSTOM";
})(CreateWatcherDtoType = exports.CreateWatcherDtoType || (exports.CreateWatcherDtoType = {}));
var CreateWatcherDto = /** @class */ (function (_super) {
    __extends(CreateWatcherDto, _super);
    function CreateWatcherDto() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "address" }),
        __metadata("design:type", String)
    ], CreateWatcherDto.prototype, "address", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "confirmationsBlocks" }),
        __metadata("design:type", Number)
    ], CreateWatcherDto.prototype, "confirmationsBlocks", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "customEventAbi" }),
        (0, class_transformer_1.Type)(function () { return CustomEventAbi; }),
        __metadata("design:type", CustomEventAbi)
    ], CreateWatcherDto.prototype, "customEventAbi", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "customFilters" }),
        (0, class_transformer_1.Type)(function () { return CustomFilters; }),
        __metadata("design:type", CustomFilters)
    ], CreateWatcherDto.prototype, "customFilters", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "description" }),
        __metadata("design:type", String)
    ], CreateWatcherDto.prototype, "description", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "metadata" }),
        (0, class_transformer_1.Type)(function () { return CreateWatcherDtoMetadata; }),
        __metadata("design:type", CreateWatcherDtoMetadata)
    ], CreateWatcherDto.prototype, "metadata", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], CreateWatcherDto.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "network" }),
        __metadata("design:type", String)
    ], CreateWatcherDto.prototype, "network", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "type" }),
        __metadata("design:type", String)
    ], CreateWatcherDto.prototype, "type", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "webhookUrl" }),
        __metadata("design:type", String)
    ], CreateWatcherDto.prototype, "webhookUrl", void 0);
    return CreateWatcherDto;
}(utils_1.SpeakeasyBase));
exports.CreateWatcherDto = CreateWatcherDto;
