// @starton/sdk/dist/sdk/models/shared/pin.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pin = exports.PinType = exports.Status = exports.PinMetadata = void 0;
var utils_1 = require("../../../internal/utils");
var directorycontent_1 = require("./directorycontent");
var class_transformer_1 = require("class-transformer");
var PinMetadata = /** @class */ (function (_super) {
    __extends(PinMetadata, _super);
    function PinMetadata() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return PinMetadata;
}(utils_1.SpeakeasyBase));
exports.PinMetadata = PinMetadata;
var Status;
(function (Status) {
    Status["Queued"] = "queued";
    Status["Pinning"] = "pinning";
    Status["Pinned"] = "pinned";
    Status["Failed"] = "failed";
    Status["Deleted"] = "deleted";
})(Status = exports.Status || (exports.Status = {}));
var PinType;
(function (PinType) {
    PinType["Directory"] = "directory";
    PinType["File"] = "file";
})(PinType = exports.PinType || (exports.PinType = {}));
var Pin = /** @class */ (function (_super) {
    __extends(Pin, _super);
    function Pin() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "cid" }),
        __metadata("design:type", String)
    ], Pin.prototype, "cid", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "createdAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], Pin.prototype, "createdAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "delegates" }),
        __metadata("design:type", Array)
    ], Pin.prototype, "delegates", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: directorycontent_1.DirectoryContent }),
        (0, class_transformer_1.Expose)({ name: "directoryContent" }),
        (0, class_transformer_1.Type)(function () { return directorycontent_1.DirectoryContent; }),
        __metadata("design:type", Array)
    ], Pin.prototype, "directoryContent", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], Pin.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "metadata" }),
        (0, class_transformer_1.Type)(function () { return PinMetadata; }),
        __metadata("design:type", PinMetadata)
    ], Pin.prototype, "metadata", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], Pin.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "origins" }),
        __metadata("design:type", Array)
    ], Pin.prototype, "origins", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "projectId" }),
        __metadata("design:type", String)
    ], Pin.prototype, "projectId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "size" }),
        __metadata("design:type", Number)
    ], Pin.prototype, "size", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "status" }),
        __metadata("design:type", String)
    ], Pin.prototype, "status", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "type" }),
        __metadata("design:type", String)
    ], Pin.prototype, "type", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "updatedAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], Pin.prototype, "updatedAt", void 0);
    return Pin;
}(utils_1.SpeakeasyBase));
exports.Pin = Pin;
