// @starton/sdk/dist/sdk/models/shared/smartcontracttemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartContractTemplate = exports.Form = exports.SmartContractTemplateCompilationDetails = exports.Category = exports.SmartContractTemplateAbi = void 0;
var utils_1 = require("../../../internal/utils");
var class_transformer_1 = require("class-transformer");
var SmartContractTemplateAbi = /** @class */ (function (_super) {
    __extends(SmartContractTemplateAbi, _super);
    function SmartContractTemplateAbi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return SmartContractTemplateAbi;
}(utils_1.SpeakeasyBase));
exports.SmartContractTemplateAbi = SmartContractTemplateAbi;
var Category;
(function (Category) {
    Category["Nft"] = "NFT";
    Category["Fungible"] = "FUNGIBLE";
    Category["Other"] = "OTHER";
    Category["Deprecated"] = "DEPRECATED";
    Category["Sale"] = "SALE";
    Category["Tool"] = "TOOL";
})(Category = exports.Category || (exports.Category = {}));
var SmartContractTemplateCompilationDetails = /** @class */ (function (_super) {
    __extends(SmartContractTemplateCompilationDetails, _super);
    function SmartContractTemplateCompilationDetails() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return SmartContractTemplateCompilationDetails;
}(utils_1.SpeakeasyBase));
exports.SmartContractTemplateCompilationDetails = SmartContractTemplateCompilationDetails;
var Form = /** @class */ (function (_super) {
    __extends(Form, _super);
    function Form() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return Form;
}(utils_1.SpeakeasyBase));
exports.Form = Form;
var SmartContractTemplate = /** @class */ (function (_super) {
    __extends(SmartContractTemplate, _super);
    function SmartContractTemplate() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: SmartContractTemplateAbi }),
        (0, class_transformer_1.Expose)({ name: "abi" }),
        (0, class_transformer_1.Type)(function () { return SmartContractTemplateAbi; }),
        __metadata("design:type", Array)
    ], SmartContractTemplate.prototype, "abi", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "blockchains" }),
        __metadata("design:type", Array)
    ], SmartContractTemplate.prototype, "blockchains", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "category" }),
        __metadata("design:type", String)
    ], SmartContractTemplate.prototype, "category", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "compilationDetails" }),
        (0, class_transformer_1.Type)(function () { return SmartContractTemplateCompilationDetails; }),
        __metadata("design:type", SmartContractTemplateCompilationDetails)
    ], SmartContractTemplate.prototype, "compilationDetails", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "createdAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], SmartContractTemplate.prototype, "createdAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "description" }),
        __metadata("design:type", String)
    ], SmartContractTemplate.prototype, "description", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "form" }),
        (0, class_transformer_1.Type)(function () { return Form; }),
        __metadata("design:type", Form)
    ], SmartContractTemplate.prototype, "form", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "githubUrl" }),
        __metadata("design:type", String)
    ], SmartContractTemplate.prototype, "githubUrl", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "humanReadableAbi" }),
        __metadata("design:type", Array)
    ], SmartContractTemplate.prototype, "humanReadableAbi", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], SmartContractTemplate.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "isActivated" }),
        __metadata("design:type", Boolean)
    ], SmartContractTemplate.prototype, "isActivated", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "isAudited" }),
        __metadata("design:type", Boolean)
    ], SmartContractTemplate.prototype, "isAudited", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], SmartContractTemplate.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "order" }),
        __metadata("design:type", Number)
    ], SmartContractTemplate.prototype, "order", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "shortDescription" }),
        __metadata("design:type", String)
    ], SmartContractTemplate.prototype, "shortDescription", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "tags" }),
        __metadata("design:type", Array)
    ], SmartContractTemplate.prototype, "tags", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "updatedAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], SmartContractTemplate.prototype, "updatedAt", void 0);
    return SmartContractTemplate;
}(utils_1.SpeakeasyBase));
exports.SmartContractTemplate = SmartContractTemplate;
