// @starton/sdk/dist/sdk/models/shared/smartcontract.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartContract = exports.SmartContractStatus = exports.State = exports.CompilationDetails = exports.Abi = void 0;
var utils_1 = require("../../../internal/utils");
var class_transformer_1 = require("class-transformer");
var Abi = /** @class */ (function (_super) {
    __extends(Abi, _super);
    function Abi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return Abi;
}(utils_1.SpeakeasyBase));
exports.Abi = Abi;
var CompilationDetails = /** @class */ (function (_super) {
    __extends(CompilationDetails, _super);
    function CompilationDetails() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return CompilationDetails;
}(utils_1.SpeakeasyBase));
exports.CompilationDetails = CompilationDetails;
var State;
(function (State) {
    State["Success"] = "SUCCESS";
    State["Pending"] = "PENDING";
    State["ManualActionRequired"] = "MANUAL_ACTION_REQUIRED";
    State["Error"] = "ERROR";
})(State = exports.State || (exports.State = {}));
var SmartContractStatus;
(function (SmartContractStatus) {
    SmartContractStatus["Unsigned"] = "UNSIGNED";
    SmartContractStatus["ErrorTx"] = "ERROR_TX";
    SmartContractStatus["ErrorPublish"] = "ERROR_PUBLISH";
    SmartContractStatus["Published"] = "PUBLISHED";
    SmartContractStatus["ReceivedByStarton"] = "RECEIVED_BY_STARTON";
    SmartContractStatus["CreatedByStarton"] = "CREATED_BY_STARTON";
    SmartContractStatus["CouldNotEstimateGasPrice"] = "COULD_NOT_ESTIMATE_GAS_PRICE";
    SmartContractStatus["CouldNotIncreaseGasPrice"] = "COULD_NOT_INCREASE_GAS_PRICE";
    SmartContractStatus["GasPriceEstimated"] = "GAS_PRICE_ESTIMATED";
    SmartContractStatus["InvalidGasPrice"] = "INVALID_GAS_PRICE";
    SmartContractStatus["ReplacementGasPriceUnderpriced"] = "REPLACEMENT_GAS_PRICE_UNDERPRICED";
    SmartContractStatus["CouldNotEstimateGasLimit"] = "COULD_NOT_ESTIMATE_GAS_LIMIT";
    SmartContractStatus["GasLimitEstimated"] = "GAS_LIMIT_ESTIMATED";
    SmartContractStatus["ExecutionWillFail"] = "EXECUTION_WILL_FAIL";
    SmartContractStatus["InvalidArgument"] = "INVALID_ARGUMENT";
    SmartContractStatus["InsufficientFunds"] = "INSUFFICIENT_FUNDS";
    SmartContractStatus["InsufficientFundsAfterBroadcast"] = "INSUFFICIENT_FUNDS_AFTER_BROADCAST";
    SmartContractStatus["CouldNotAssignNonce"] = "COULD_NOT_ASSIGN_NONCE";
    SmartContractStatus["CouldNotUnstuckNonce"] = "COULD_NOT_UNSTUCK_NONCE";
    SmartContractStatus["NonceAssigned"] = "NONCE_ASSIGNED";
    SmartContractStatus["NonceExpired"] = "NONCE_EXPIRED";
    SmartContractStatus["CouldNotSign"] = "COULD_NOT_SIGN";
    SmartContractStatus["Signed"] = "SIGNED";
    SmartContractStatus["SentToMempool"] = "SENT_TO_MEMPOOL";
    SmartContractStatus["CouldNotBroadcast"] = "COULD_NOT_BROADCAST";
    SmartContractStatus["AlreadyKnown"] = "ALREADY_KNOWN";
    SmartContractStatus["Mined"] = "MINED";
    SmartContractStatus["Confirmed"] = "CONFIRMED";
    SmartContractStatus["Replaced"] = "REPLACED";
    SmartContractStatus["Failed"] = "FAILED";
    SmartContractStatus["MonitoringInProgress"] = "MONITORING_IN_PROGRESS";
    SmartContractStatus["StuckByPreviousTransaction"] = "STUCK_BY_PREVIOUS_TRANSACTION";
    SmartContractStatus["MaxGasPriceReach"] = "MAX_GAS_PRICE_REACH";
    SmartContractStatus["GasPriceIncreased"] = "GAS_PRICE_INCREASED";
    SmartContractStatus["NewTransactionHash"] = "NEW_TRANSACTION_HASH";
    SmartContractStatus["Unknown"] = "UNKNOWN";
    SmartContractStatus["MonitoringInterrupted"] = "MONITORING_INTERRUPTED";
})(SmartContractStatus = exports.SmartContractStatus || (exports.SmartContractStatus = {}));
var SmartContract = /** @class */ (function (_super) {
    __extends(SmartContract, _super);
    function SmartContract() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: Abi }),
        (0, class_transformer_1.Expose)({ name: "abi" }),
        (0, class_transformer_1.Type)(function () { return Abi; }),
        __metadata("design:type", Array)
    ], SmartContract.prototype, "abi", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "address" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "address", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "blockNumber" }),
        __metadata("design:type", Number)
    ], SmartContract.prototype, "blockNumber", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "compilationDetails" }),
        (0, class_transformer_1.Type)(function () { return CompilationDetails; }),
        __metadata("design:type", CompilationDetails)
    ], SmartContract.prototype, "compilationDetails", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "createdAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], SmartContract.prototype, "createdAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "creationHash" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "creationHash", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "description" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "description", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "minedDate" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], SmartContract.prototype, "minedDate", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "network" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "network", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "params" }),
        __metadata("design:type", Array)
    ], SmartContract.prototype, "params", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "projectId" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "projectId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "state" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "state", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "status" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "status", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "templateId" }),
        __metadata("design:type", String)
    ], SmartContract.prototype, "templateId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "updatedAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], SmartContract.prototype, "updatedAt", void 0);
    return SmartContract;
}(utils_1.SpeakeasyBase));
exports.SmartContract = SmartContract;
