// @starton/sdk/dist/sdk/models/shared/watcher.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watcher = exports.WatcherType = exports.TriggerType = exports.WatcherMetadata = exports.WatcherCustomFilters = exports.WatcherCustomEventAbi = void 0;
var utils_1 = require("../../../internal/utils");
var class_transformer_1 = require("class-transformer");
var WatcherCustomEventAbi = /** @class */ (function (_super) {
    __extends(WatcherCustomEventAbi, _super);
    function WatcherCustomEventAbi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return WatcherCustomEventAbi;
}(utils_1.SpeakeasyBase));
exports.WatcherCustomEventAbi = WatcherCustomEventAbi;
var WatcherCustomFilters = /** @class */ (function (_super) {
    __extends(WatcherCustomFilters, _super);
    function WatcherCustomFilters() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return WatcherCustomFilters;
}(utils_1.SpeakeasyBase));
exports.WatcherCustomFilters = WatcherCustomFilters;
var WatcherMetadata = /** @class */ (function (_super) {
    __extends(WatcherMetadata, _super);
    function WatcherMetadata() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return WatcherMetadata;
}(utils_1.SpeakeasyBase));
exports.WatcherMetadata = WatcherMetadata;
var TriggerType;
(function (TriggerType) {
    TriggerType["Block"] = "BLOCK";
    TriggerType["Transaction"] = "TRANSACTION";
    TriggerType["Event"] = "EVENT";
    TriggerType["GasPrice"] = "GAS_PRICE";
})(TriggerType = exports.TriggerType || (exports.TriggerType = {}));
var WatcherType;
(function (WatcherType) {
    WatcherType["MinedTransaction"] = "MINED_TRANSACTION";
    WatcherType["AddressActivity"] = "ADDRESS_ACTIVITY";
    WatcherType["AddressReceivedNativeCurrency"] = "ADDRESS_RECEIVED_NATIVE_CURRENCY";
    WatcherType["AddressSentNativeCurrency"] = "ADDRESS_SENT_NATIVE_CURRENCY";
    WatcherType["EventTransfer"] = "EVENT_TRANSFER";
    WatcherType["Erc721EventTransfer"] = "ERC721_EVENT_TRANSFER";
    WatcherType["Erc1155EventTransferSingle"] = "ERC1155_EVENT_TRANSFER_SINGLE";
    WatcherType["Erc1155EventTransferBatch"] = "ERC1155_EVENT_TRANSFER_BATCH";
    WatcherType["EventMint"] = "EVENT_MINT";
    WatcherType["EventApproval"] = "EVENT_APPROVAL";
    WatcherType["EventCustom"] = "EVENT_CUSTOM";
})(WatcherType = exports.WatcherType || (exports.WatcherType = {}));
var Watcher = /** @class */ (function (_super) {
    __extends(Watcher, _super);
    function Watcher() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "address" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "address", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "confirmationsBlocks" }),
        __metadata("design:type", Number)
    ], Watcher.prototype, "confirmationsBlocks", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "createdAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], Watcher.prototype, "createdAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: WatcherCustomEventAbi }),
        (0, class_transformer_1.Expose)({ name: "customEventAbi" }),
        (0, class_transformer_1.Type)(function () { return WatcherCustomEventAbi; }),
        __metadata("design:type", Array)
    ], Watcher.prototype, "customEventAbi", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "customFilters" }),
        (0, class_transformer_1.Type)(function () { return WatcherCustomFilters; }),
        __metadata("design:type", WatcherCustomFilters)
    ], Watcher.prototype, "customFilters", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "description" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "description", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "id" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "id", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "metadata" }),
        (0, class_transformer_1.Type)(function () { return WatcherMetadata; }),
        __metadata("design:type", WatcherMetadata)
    ], Watcher.prototype, "metadata", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "name" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "network" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "network", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "paused" }),
        __metadata("design:type", Boolean)
    ], Watcher.prototype, "paused", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "pausedReason" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "pausedReason", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "projectId" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "projectId", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "triggerType" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "triggerType", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "type" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "type", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "updatedAt" }),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return new Date(value);
        }, { toClassOnly: true }),
        __metadata("design:type", Date)
    ], Watcher.prototype, "updatedAt", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "webhookUrl" }),
        __metadata("design:type", String)
    ], Watcher.prototype, "webhookUrl", void 0);
    return Watcher;
}(utils_1.SpeakeasyBase));
exports.Watcher = Watcher;
