// @starton/sdk/dist/sdk/models/operations/getallsmartcontracttemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAllSmartContractTemplateResponse = exports.GetAllSmartContractTemplateSmartContractTemplatePaginated = exports.GetAllSmartContractTemplateRequest = exports.Category = void 0;
var utils_1 = require("../../../internal/utils");
var shared = __importStar(require("../../../sdk/models/shared"));
var class_transformer_1 = require("class-transformer");
var Category;
(function (Category) {
    Category["Nft"] = "NFT";
    Category["Fungible"] = "FUNGIBLE";
    Category["Other"] = "OTHER";
    Category["Deprecated"] = "DEPRECATED";
    Category["Sale"] = "SALE";
    Category["Tool"] = "TOOL";
})(Category = exports.Category || (exports.Category = {}));
var GetAllSmartContractTemplateRequest = /** @class */ (function (_super) {
    __extends(GetAllSmartContractTemplateRequest, _super);
    function GetAllSmartContractTemplateRequest() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=blockchain" }),
        __metadata("design:type", String)
    ], GetAllSmartContractTemplateRequest.prototype, "blockchain", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=category" }),
        __metadata("design:type", String)
    ], GetAllSmartContractTemplateRequest.prototype, "category", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({
            data: "queryParam, style=form;explode=true;name=includeCompilationDetails",
        }),
        __metadata("design:type", Boolean)
    ], GetAllSmartContractTemplateRequest.prototype, "includeCompilationDetails", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=includeForm" }),
        __metadata("design:type", Boolean)
    ], GetAllSmartContractTemplateRequest.prototype, "includeForm", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=isActivated" }),
        __metadata("design:type", Boolean)
    ], GetAllSmartContractTemplateRequest.prototype, "isActivated", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=isAudited" }),
        __metadata("design:type", Boolean)
    ], GetAllSmartContractTemplateRequest.prototype, "isAudited", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=limit" }),
        __metadata("design:type", Number)
    ], GetAllSmartContractTemplateRequest.prototype, "limit", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=name" }),
        __metadata("design:type", String)
    ], GetAllSmartContractTemplateRequest.prototype, "name", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ data: "queryParam, style=form;explode=true;name=page" }),
        __metadata("design:type", Number)
    ], GetAllSmartContractTemplateRequest.prototype, "page", void 0);
    return GetAllSmartContractTemplateRequest;
}(utils_1.SpeakeasyBase));
exports.GetAllSmartContractTemplateRequest = GetAllSmartContractTemplateRequest;
var GetAllSmartContractTemplateSmartContractTemplatePaginated = /** @class */ (function (_super) {
    __extends(GetAllSmartContractTemplateSmartContractTemplatePaginated, _super);
    function GetAllSmartContractTemplateSmartContractTemplatePaginated() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)({ elemType: shared.SmartContractTemplate }),
        (0, class_transformer_1.Expose)({ name: "items" }),
        (0, class_transformer_1.Type)(function () { return shared.SmartContractTemplate; }),
        __metadata("design:type", Array)
    ], GetAllSmartContractTemplateSmartContractTemplatePaginated.prototype, "items", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        (0, class_transformer_1.Expose)({ name: "meta" }),
        (0, class_transformer_1.Type)(function () { return shared.PaginationData; }),
        __metadata("design:type", shared.PaginationData)
    ], GetAllSmartContractTemplateSmartContractTemplatePaginated.prototype, "meta", void 0);
    return GetAllSmartContractTemplateSmartContractTemplatePaginated;
}(utils_1.SpeakeasyBase));
exports.GetAllSmartContractTemplateSmartContractTemplatePaginated = GetAllSmartContractTemplateSmartContractTemplatePaginated;
var GetAllSmartContractTemplateResponse = /** @class */ (function (_super) {
    __extends(GetAllSmartContractTemplateResponse, _super);
    function GetAllSmartContractTemplateResponse() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", String)
    ], GetAllSmartContractTemplateResponse.prototype, "contentType", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", GetAllSmartContractTemplateSmartContractTemplatePaginated)
    ], GetAllSmartContractTemplateResponse.prototype, "smartContractTemplatePaginated", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", Number)
    ], GetAllSmartContractTemplateResponse.prototype, "statusCode", void 0);
    __decorate([
        (0, utils_1.SpeakeasyMetadata)(),
        __metadata("design:type", Object)
    ], GetAllSmartContractTemplateResponse.prototype, "rawResponse", void 0);
    return GetAllSmartContractTemplateResponse;
}(utils_1.SpeakeasyBase));
exports.GetAllSmartContractTemplateResponse = GetAllSmartContractTemplateResponse;
